File: 1701-average-waiting-time/1701-average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double ans;
        int currCompletion=customers[0][0]+customers[0][1];
        ans=currCompletion-customers[0][0];
        int n=customers.size();
        for(
int i=1;
i<n;
i++
        ){
           currCompletion+= currCompletion<customers[i][0]?customers[i][0]-currCompletion:0;
            currCompletion=customers[i][1]+currCompletion;
             ans+=currCompletion-customers[i][0];

        }
        return ((double)ans/n);
    }
};

File: 1701-average-waiting-time/1701-average-waiting-time.java
class Solution {
public double averageWaitingTime(int[][] customers) {
long timestamp = 0, sum = 0;
for(int[] customer : customers){
if(timestamp<customer[0]) timestamp =timestamp+ (customer[0]-timestamp);
timestamp+=customer[1];
sum+= timestamp - customer[0];
}
return (double) sum/customers.length;
}
}